src/search3_indexer.erl

```diff
@@ -0,0 +1,221 @@
+-module(search3_indexer).
+
+-export([
+    update/2
+]).
+
+-export([
+    spawn_link/0
+]).
+
+
+-export([
+    init/0
+]).
+
+-include("search3.hrl").
+-include_lib("couch/include/couch_db.hrl").
+-include_lib("fabric/include/fabric2.hrl").
+-include_lib("eunit/include/eunit.hrl").
+
+-import(couch_query_servers, [
+    get_os_process/1,
+    ret_os_process/1,
+    proc_prompt/2
+]).
+
+spawn_link() ->
+    proc_lib:spawn_link(?MODULE, init, []).
+
+init() ->
+    {ok, Job, JobData} = couch_jobs:accept(?SEARCH_JOB_TYPE, #{}),
+    #{
+        <<"db_name">> := DbName,
+        <<"ddoc_id">> := DDocId,
+        <<"name">> := IndexName,
+        <<"sig">> := JobSig
+    } = JobData,
+    {ok, Db} = fabric2_db:open(DbName, []),
+    {ok, DDoc} = fabric2_db:open_doc(Db, DDocId),
+    {ok, Index} = search3_util:design_doc_to_index(DDoc, IndexName),
+    Index1 = Index#index{dbname = DbName},
+    HexSig = fabric2_util:to_hex(Index1#index.sig),
+
+    if  HexSig == JobSig -> ok; true ->
+        couch_jobs:finish(undefined, Job, JobData#{
+            error => sig_changed,
```

Comment:
Here, keep in mind that couch_jobs serializes JobData into json. So all the atom keys will be decoded to binary when read back. Mentioning in case you had places where you read from JobData and match on `error` instead of `<<"error">>`. It might be better to stick to binaries in general.  (Could defined macros maybe to make it consistent).